crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -1421,6 +1420,22 @@ impl Analyzer<'_, '_> {
         }
 
         match rhs {
+            Type::Interface(Interface {
+                span,
+                name,
+                type_params,
+                extends,
+                body,
+                metadata: InterfaceMetadata { common },
+                tracker,
+            }) => {
+                for parent in extends {
+                    let ty = self.type_of_ts_entity_name(parent.span(), &parent.expr, parent.type_args.as_deref())?;
+                    if let Type::Array(_) = &ty {
```

Comment:
Checking for `Type::Array` is not enough, and we should handle this by accessing iterating over properties

Reply:
```ts
interface MyTuple extends Array<unknown> {
    0: number
    1: number
}

declare let a: MyTuple
declare let b: [number, number]

a = b // ok
b = a // error
```

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgLIE8AqBXADgGxQgA9IQATAZ2QEEoo50AebEAaxAHsB3EAPmQBvAFDIxyAAwAuZCGwBbAEbRR4gIwy5SlQF9hw8hAT44UFITDI4MjDgIQDRk2eQXkimQG0tyqABpZBV8AXX04ZABed2QAehjkTjZhRUirWPjoKE4oIA

Reply:
```ts
interface MyTuple extends Array<any> {
    0: number
    1: number
}

declare let a: MyTuple
declare let b: [number, number]

a = b // ok
b = a // error
```

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgLIE8AqBXADgGxQgA9IQATAZ2QEEoo50AeOEdAPmQG8AoZf5AAYAXMhDYAtgCNofAQEZR46bIC+PHuQgJ8cKCkJhkcURhwEIm7bv3JDyKaIDaymVAA0YyW4C6GuMgAvA7IAPShyAD2ANY8UkHGYRHQUJFQQA

Reply:
As you can see, we can't rely on the type argument in `extends Array<T>`